browser.js

```diff
@@ -311,8 +325,17 @@ Socket.prototype.connect = function(options, cb) {
 				};
 			}
 
-			if (data.error) {
-				self.emit('error', 'Cannot open TCP connection ['+res.statusCode+']: '+data.error);
+			if (data.error !== undefined) {
```

Comment:
it could be an empty string (so this is a bugfix), also added a few useful explanations